Core/tests/ExpressionParser2.cpp

```diff
@@ -194,64 +226,83 @@ TEST_CASE("ExpressionParser2", "[common][events]") {
       REQUIRE(validator.GetErrors()[0]->GetStartPosition() == 12);
     }
     {
-      auto node = parser.ParseExpression("string", "\"\"\"");
+      auto node = parser.ParseExpression("\"\"\"");
       REQUIRE(node != nullptr);
 
-      gd::ExpressionValidator validator;
+      gd::ExpressionValidator validator(platform, project, layout1, "string");
       node->Visit(validator);
-      REQUIRE(validator.GetErrors().size() == 2);
+      REQUIRE(validator.GetErrors().size() == 3);
       REQUIRE(validator.GetErrors()[0]->GetMessage() ==
+              "More than one term was found. Verify that your expression is "
+              "properly written.");
+      REQUIRE(validator.GetErrors()[1]->GetMessage() ==
               "You must add the operator + between texts or expressions. For "
               "example: \"Your name: \" + VariableString(PlayerName).");
-      REQUIRE(validator.GetErrors()[0]->GetStartPosition() == 2);
-      REQUIRE(validator.GetErrors()[1]->GetMessage() ==
+      REQUIRE(validator.GetErrors()[1]->GetStartPosition() == 2);
+      REQUIRE(validator.GetErrors()[2]->GetMessage() ==
               "A text must end with a double quote (\"). Add a double quote to "
               "terminate the text.");
     }
   }
 
   SECTION("Unterminated expressions/extra characters") {
     {
-      auto node = parser.ParseExpression("string", "\"hello\",");
+      auto node = parser.ParseExpression("\"hello\",");
       REQUIRE(node != nullptr);
 
-      gd::ExpressionValidator validator;
+      gd::ExpressionValidator validator(platform, project, layout1, "string");
       node->Visit(validator);
-      REQUIRE(validator.GetErrors().size() == 1);
+      REQUIRE(validator.GetErrors().size() == 3);
+
+      // TODO Find a way to remove these 2 extra errors.
       REQUIRE(validator.GetErrors()[0]->GetMessage() ==
+              "You must add the operator + between texts or expressions. For "
+              "example: \"Your name: \" + VariableString(PlayerName).");
+      REQUIRE(validator.GetErrors()[2]->GetMessage() ==
+              "You must enter a text (between quotes) or a valid expression call.");
+
+      REQUIRE(validator.GetErrors()[1]->GetMessage() ==
               "The expression has extra character at the end that should be "
               "removed (or completed if your expression is not finished).");
     }
```

Comment:
These 2 extra errors happen because the validation is done on the tree without checking that the empty node at the right of the "," operator is actually an unfinished expression.
These errors could be pertinent but 3 errors might be a bit too much. Should this case be handled to remove these 2 errors?